src/drawGeofences.ts

```diff
@@ -0,0 +1,128 @@
+import { Feature } from "geojson";
+import { Map as maplibreMap } from "maplibre-gl";
+import { isGeofence } from "./utils";
+import { Geofence, Polygon } from "./types";
+import { COLOR_BLACK } from "./constants";
+
+const FILL_OPACITY = 0.3;
+const BORDER_OPACITY = 0.5;
+const BORDER_WIDTH = 4;
+
+/**
+ */
+export interface DrawGeofencesOptions {
+  fillColor?: string;
+  fillOpacity?: number;
+  borderColor?: string;
+  borderWidth?: number;
+  borderOpacity?: number;
+}
+
+export interface DrawGeofencesOutput {
+  sourceId: string;
+  outlineLayerId: string;
+  fillLayerId: string;
+  show: () => void;
+  hide: () => void;
+}
+
+/**
+ * DrawPoints utility function for adding points to a map based on coordinate data or a FeatureCollection. Will add clustered points and styled markers by default with options for popups and other styles
+ */
+export function drawGeofences(
+  sourceName: string,
+  data: Polygon[] | Geofence[],
+  map: maplibreMap,
+  options: DrawGeofencesOptions = {}
```

Comment:
Should this be an optional argument?

Reply:
It is optional! the assignment to an empty {} makes it have a default param